### PR TITLE
[WIP] Enhancement/opentracing

### LINKED ISSUE #498: Add OpenTracing to the deployments
https://github.com/microservices-demo/microservices-demo/issues/479

### PR DESCRIPTION
**Pull request by jasonrichardsmith**
resolves #498 for catalogue and user only

### CODE REVIEW

File: deploy/docker-compose/docker-compose.yml [hunk 1]
```diff
@@ -193,3 +196,20 @@ services:
...
+      - all
+    cap_add:
+      - CHOWN
+      - SETGID
+      - SETUID
+    read_only: true
+    tmpfs:
+      - /tmp:rw,noexec,nosuid
+    environment:
+      - reschedule=on-node-failure
```
Comment (philwinder) [inline deploy/docker-compose/docker-compose.yml:213]: AFAIK, reschedule is only for docker swarm.

File: deploy/docker-compose-weave/docker-compose.yml [hunk 2]
```diff
@@ -160,6 +163,23 @@ services:
     networks:
       - external
     command: "-d 60 -r 200 -c 2 -h edge-router"
+  zipkin:
```
Comment (philwinder) [inline deploy/docker-compose-weave/docker-compose.yml:166]: Not attached to any network. Is that intentional?

File: deploy/kubernetes/manifests/catalogue-dep.yaml [hunk 3]
```diff
@@ -16,7 +16,10 @@ spec:
       containers:
       - name: catalogue
         image: weaveworksdemos/catalogue:0.2.0
```
Comment (philwinder) [inline deploy/kubernetes/manifests/catalogue-dep.yaml:18]: Wrong catalogue version

**Review by philwinder — CHANGES_REQUESTED** [review]

**Comment by jasonrichardsmith** [comment]
Closed, each deployment got its own PR.

**Pull request closed without merging**